platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/utils/UserTransformer.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.user.server.utils;
+
+import org.eclipse.che.api.core.rest.ServiceContext;
+import org.eclipse.che.api.core.rest.shared.dto.Link;
+import org.eclipse.che.api.core.util.LinksHelper;
+import org.eclipse.che.api.user.server.UserProfileService;
+import org.eclipse.che.api.user.server.UserService;
+import org.eclipse.che.api.user.server.dao.User;
+import org.eclipse.che.api.user.shared.dto.UserDescriptor;
+import org.eclipse.che.dto.server.DtoFactory;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriBuilder;
+import java.util.LinkedList;
+import java.util.List;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_CURRENT_USER;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_CURRENT_USER_PROFILE;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_BY_EMAIL;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_PROFILE_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_REMOVE_USER_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_UPDATE_PASSWORD;
+
+/**
+ * @author Anatoliy Bazko
+ */
+public class UserTransformer {
+
```

Comment:
Make this class similar to https://github.com/codenvy/che-core/blob/master/platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/DtoConverter.java